### PR TITLE
Add prometheus metrics output

### PR DESCRIPTION
**Pull request by crosbymichael**
The interval that metrics are collected on is based on the prometheus interval.  Metrics are collected when a request hits the metrics endpoint. 

```
containerd_container_blkio_io_service_bytes_recursive_bytes{id="test",major="8",minor="0",op="Async"} 950272
containerd_container_blkio_io_service_bytes_recursive_bytes{id="test",major="8",minor="0",op="Read"} 950272
containerd_container_blkio_io_service_bytes_recursive_bytes{id="test",major="8",minor="0",op="Sync"} 0
containerd_container_blkio_io_service_bytes_recursive_bytes{id="test",major="8",minor="0",op="Total"} 950272
containerd_container_blkio_io_service_bytes_recursive_bytes{id="test",major="8",minor="0",op="Write"} 0
containerd_container_blkio_io_serviced_recursive_total{id="test",major="8",minor="0",op="Async"} 15
containerd_container_blkio_io_serviced_recursive_total{id="test",major="8",minor="0",op="Read"} 15
containerd_container_blkio_io_serviced_recursive_total{id="test",major="8",minor="0",op="Sync"} 0
containerd_container_blkio_io_serviced_recursive_total{id="test",major="8",minor="0",op="Total"} 15
containerd_container_blkio_io_serviced_recursive_total{id="test",major="8",minor="0",op="Write"} 0
containerd_container_cpu_kernel_nanoseconds{id="test"} 0
containerd_container_cpu_throttle_periods_total{id="test"} 0
containerd_container_cpu_throttled_periods_total{id="test"} 0
containerd_container_cpu_throttled_time_nanoseconds{id="test"} 0
containerd_container_cpu_total_nanoseconds{id="test"} 1.432225e+07
containerd_container_cpu_user_nanoseconds{id="test"} 0
containerd_container_hugetlb_failcnt_total{id="test",page="1GB"} 0
containerd_container_hugetlb_failcnt_total{id="test",page="2MB"} 0
containerd_container_hugetlb_max_bytes{id="test",page="1GB"} 0
containerd_container_hugetlb_max_bytes{id="test",page="2MB"} 0
containerd_container_hugetlb_usage_bytes{id="test",page="1GB"} 0
containerd_container_hugetlb_usage_bytes{id="test",page="2MB"} 0
containerd_container_memory_active_anon_bytes{id="test"} 73728
containerd_container_memory_active_file_bytes{id="test"} 643072
containerd_container_memory_cache_bytes{id="test"} 950272
containerd_container_memory_dirty_bytes{id="test"} 0
containerd_container_memory_hierarchical_memory_limit_bytes{id="test"} 9.223372036854772e+18
containerd_container_memory_hierarchical_memsw_limit_bytes{id="test"} 9.223372036854772e+18
containerd_container_memory_inactive_anon_bytes{id="test"} 0
containerd_container_memory_inactive_file_bytes{id="test"} 307200
containerd_container_memory_kernel_failcnt_total{id="test"} 0
containerd_container_memory_kernel_limit_bytes{id="test"} 9.223372036854772e+18
containerd_container_memory_kernel_max_bytes{id="test"} 0
containerd_container_memory_kernel_usage_bytes{id="test"} 0
containerd_container_memory_kerneltcp_failcnt_total{id="test"} 0
containerd_container_memory_kerneltcp_limit_bytes{id="test"} 9.223372036854772e+18
containerd_container_memory_kerneltcp_max_bytes{id="test"} 0
containerd_container_memory_kerneltcp_usage_bytes{id="test"} 0
containerd_container_memory_mapped_file_bytes{id="test"} 516096
containerd_container_memory_oom_total{id="test"} 0
containerd_container_memory_pgfault_bytes{id="test"} 503
containerd_container_memory_pgmajfault_bytes{id="test"} 6
containerd_container_memory_pgpgin_bytes{id="test"} 516
containerd_container_memory_pgpgout_bytes{id="test"} 266
containerd_container_memory_rss_bytes{id="test"} 73728
containerd_container_memory_rss_huge_bytes{id="test"} 0
containerd_container_memory_swap_failcnt_total{id="test"} 0
containerd_container_memory_swap_limit_bytes{id="test"} 9.223372036854772e+18
containerd_container_memory_swap_max_bytes{id="test"} 1.41312e+06
containerd_container_memory_swap_usage_bytes{id="test"} 1.024e+06
containerd_container_memory_total_active_anon_bytes{id="test"} 73728
containerd_container_memory_total_active_file_bytes{id="test"} 643072
containerd_container_memory_total_cache_bytes{id="test"} 950272
containerd_container_memory_total_dirty_bytes{id="test"} 0
containerd_container_memory_total_inactive_anon_bytes{id="test"} 0
containerd_container_memory_total_inactive_file_bytes{id="test"} 307200
containerd_container_memory_total_mapped_file_bytes{id="test"} 516096
containerd_container_memory_total_pgfault_bytes{id="test"} 503
containerd_container_memory_total_pgmajfault_bytes{id="test"} 6
containerd_container_memory_total_pgpgin_bytes{id="test"} 516
containerd_container_memory_total_pgpgout_bytes{id="test"} 266
containerd_container_memory_total_rss_bytes{id="test"} 73728
containerd_container_memory_total_rss_huge_bytes{id="test"} 0
containerd_container_memory_total_unevictable_bytes{id="test"} 0
containerd_container_memory_total_writeback_bytes{id="test"} 0
containerd_container_memory_unevictable_bytes{id="test"} 0
containerd_container_memory_usage_failcnt_total{id="test"} 0
containerd_container_memory_usage_limit_bytes{id="test"} 9.223372036854772e+18
containerd_container_memory_usage_max_bytes{id="test"} 1.41312e+06
containerd_container_memory_usage_usage_bytes{id="test"} 1.024e+06
containerd_container_memory_writeback_bytes{id="test"} 0
containerd_container_per_cpu_nanoseconds{cpu="0",id="test"} 6.12895e+06
containerd_container_per_cpu_nanoseconds{cpu="1",id="test"} 1.931172e+06
containerd_container_per_cpu_nanoseconds{cpu="2",id="test"} 4.067339e+06
containerd_container_per_cpu_nanoseconds{cpu="3",id="test"} 2.194789e+06
containerd_container_pids_current{id="test"} 1
containerd_container_pids_limit{id="test"} 0
```

### CODE REVIEW

**Comment by crosbymichael** [comment]
There are currently 77 metrics per container.

File: prometheus/oom.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	defer o.mu.Unlock()
+	fd, err := cg.OOMEventFD()
+	if err != nil {
+		return err
+	}
+	o.set[fd] = &oom{
+		id: id,
+		c:  cg,
+	}
+	// set the gague's default value
```
Comment (dmcgowan) [inline prometheus/oom.go]: gauge's

**Comment by dmcgowan** [comment]
LGTM

**Comment by rogaha** [comment]
👍

**Pull request merged**